### PR TITLE
Set password mobile

### LINKED ISSUE #577: Implement Set Password mobile screen
https://invis.io/6NNS8Z88J3Z
https://invis.io/E3NS8ZU5XZH

### LINKED ISSUE #578: Implement Password Confirm mobile screen
https://invis.io/J4NS90CVECG

### PR DESCRIPTION
**Pull request by tanx**
Closes #577, closes #578

### CODE REVIEW

File: src/action/wallet.js [hunk 1]
```diff
@@ -202,11 +240,11 @@ class WalletAction {
    * there was no typo.
    * @return {Promise<undefined>}
    */
-  async checkNewPassword() {
+  async checkNewPassword(minLength = MIN_PASSWORD_LENGTH) {
```
[the remark this answers is not in the document]
Comment (tanx) [reply src/action/wallet.js]: Yeah notification aren't wired up yet. Currently working on https://github.com/lightninglabs/lightning-app/issues/607 next to handle PIN screen errors.

File: src/component/pin-entry.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+//
+
+const pinEntryStyles = StyleSheet.create({
+  wrapper: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+});
+
+export const PinBubbles = ({ pin }) => (
```
[the remark this answers is not in the document]
Comment (tanx) [reply src/component/pin-entry.js]: The mobile UI is currently only tested on phone resolutions. I'll share screens in app-dev.

File: src/action/wallet.js [hunk 3]
```diff
@@ -117,6 +118,43 @@ class WalletAction {
...
+  /**
+   * Remove the last digit from the password parameter.
+   * @param  {string} options.param The password parameter name
+   * @return {undefined}
+   */
+  popPinDigit({ param }) {
+    const { wallet } = this._store;
+    if (wallet[param]) {
+      wallet[param] = wallet[param].slice(0, -1);
+    } else if (param === 'passwordVerify') {
```
[the remark this answers is not in the document]
Comment (tanx) [reply src/action/wallet.js]: The unlock screen is a different view implemented in `password-mobile.js` ... coming up on the next branch. I agree we could add a back button. That would be a bit of a deviation from the current designs though and probably better to do that in another PR.

File: src/component/pin-entry.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  column: {
+    flex: 1,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+});
+
+export const PinKeyboard = ({ onInput, onBackspace }) => (
```
[the remark this answers is not in the document]
Comment (tanx) [reply src/component/pin-entry.js]: Did you test on Android? Currently only testing on iOS.

**Comment by tanx** [comment]
> Love the way these screens turned out :) I pushed a small change that allows navigation to Home after pin setup, but if your PR that builds on top of this already does that feel free to remove the commit.
> 
> Seems like mobile is really coming together! :D

@valentinewallace thanks. The change looks good to me. I agree with your comments. Those should all be addressed in future PRs.

File: src/component/pin-entry.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (valentinewallace) [reply src/component/pin-entry.js]: Yeah only on Android, if it's fast enough on iOS that should be fine

**Pull request merged**